cs/src/core/ClientSession/ClientSession.cs

```diff
@@ -747,13 +747,53 @@ public void ConcurrentReader(ref Key key, ref Input input, ref Value value, ref
 
             public bool ConcurrentWriter(ref Key key, ref Value src, ref Value dst, long address)
             {
-                return _clientSession.functions.ConcurrentWriter(ref key, ref src, ref dst);
+                if (!_clientSession.fht.SupportsMutableIndexes)
+                    return _clientSession.functions.ConcurrentWriter(ref key, ref src, ref dst);
+
+                ref RecordInfo recordInfo = ref _clientSession.fht.RecordAccessor.SpinLockRecordInfo(address);
```

Comment:
It's a bit worrisome that locking is all over the place and using spin locks.
What if the key or value were 8 byte longs and we were talking to a KeyIndex so we didnt need a lock at all?
What if the user had space in key/value to have a better implementation of lock.
Would it be better if we made this a duty of functions - capable of being overridden by specific functions. like `functions.LockRecord(ref RecordInfo, ref Key, ref Value)`.